rewrite-core/src/main/java/org/openrewrite/config/DeclarativeRecipe.java

```diff
@@ -70,13 +70,13 @@ public boolean causesAnotherCycle() {
     private final List<Recipe> preconditions = new ArrayList<>();
 
     public void addPrecondition(Recipe recipe) {
-        preconditions.add(recipe);
+        uninitializedPreconditions.add(recipe);
```

Comment:
Where is this method even called? In my `rewrite` repo it is only called from a test.

Reply:
Can you double check that we really need to change this? Because I get the impression that then `preconditions` would always be empty, which seems weird given the logic that was added in `getRecipeList()` to support preconditions in declarative recipes.

Reply:
you are right, it's only called from a test, but had to do this change to keep the test working 😅 
Probably it was also being used when on yaml declarative recipes we could have preconditions on the yaml definition?

Reply:
When we call `initialize(uninitializedPreconditions, preconditions, availableRecipes, recipeToContributors);` on line 90, it will copy and initialize (if needed) `preconditions` from `uninitializedPreconditions`.